services/src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -182,7 +184,7 @@ public void run() {
           if (topicState.replicationFactor() == -1) {
             return;
           }
-          if (topicState.someBrokerMissingPartition()) {
+          if (topicState.someBrokerMissingPartition() && !topicAssignmentIsRunning()) {
```

Comment:
Why do we need to wait for topic assignment to complete or even checking if there is ongoing assignment? Even if we do it, we may still see ZkNodeExistsException when we call `reassignPartitions(..)`. We can probably simply do the following:

```
 if (topicState.someBrokerMissingPartition())
  reassignPartitions(topicState._allBrokers, topicState._partitionInfo.size(), topicState.replicationFactor());
```